### ISSUE TITLE
How to get forecast information for every draw from PyDLM?

### ISSUE THREAD

**Issue by Simha-Kalimipalli** [wwrechard/PyDLM#69]
Hi There,
Bayesian time series models typically return a 2d array of the forecast for every draw and every timestamp that is predicted. I skimmed through the documentation and I was not able to find a class method to get that info from my Pydlm model. Is there any way to retrieve this array from pydlm?

Thanks

**Comment by wwrechard** [comment]
Hi @Simha-Kalimipalli, this package is specialized for fast time series computation. Because of that goal, it used the Kalman filter to derive the marginal distribution at each timestamp (which is a Gaussian distribution because of the normal-normal prior assumption) instead of sampling the entire posterior series. You will be able to find the mean and variance of the marginal distribution using the provided APIs within the package.

However, if you do want to sample the posterior sequence of the fitted time series rather the marginal distribution at each step, then you will need to look at other packages (e.g., [pybats](https://github.com/lavinei/pybats) which was implemented by the student of Dr. West). I did implement the [backward sampling](https://github.com/wwrechard/pydlm/blob/master/pydlm/base/kalmanFilter.py#L241) algorithm in this package but it never got used/exposed in the latter development because of the slowness and most people mostly care about marginal distribution for inferencing.
